utility/serial.cpp

```diff
@@ -92,6 +92,71 @@ void ArdusatSerial::begin(unsigned long baud, bool setXbeeSpeed)
   }
 }
 
+/**
+ * Begin serial communications with a Sparkfun BlueSMiRF module at the specified baud
+ * rate.
+ *
+ * Note that baud rates above ~57600 are not well-supported by SoftwareSerial, and 
+ * even 57600 may cause some bugs.
+ *
+ * @param baud rate for serial communications
+ *
+ * NOTE: If this function is called multiple times with different values for 'baud'
+ *       without losing power in between calls, the baud rate will not be updated
+ *       after the first call.
+ */
+void ArdusatSerial::beginBluetooth(unsigned long baud)
+{
+  bool validBaud = true;
+  char baudCmd[8];
+
+  // Temporarily sets baud rate until power loss
+  switch (baud) {
+    case 1200:  strcpy(baudCmd, "U,1200,N"); break;
+    case 2400:  strcpy(baudCmd, "U,2400,N"); break;
+    case 4800:  strcpy(baudCmd, "U,4800,N"); break;
+    case 9600:  strcpy(baudCmd, "U,9600,N"); break;
+    case 19200: strcpy(baudCmd, "U,192K,N"); break;
+    case 38400: strcpy(baudCmd, "U,384K,N"); break;
+    case 57600: strcpy(baudCmd, "U,576K,N"); break;
+    default:    validBaud = false;
+  }
+
+  if (validBaud) {
+    if (_mode == SERIAL_MODE_HARDWARE) {
+      if (baud != 9600) {
+        Serial.begin(9600);   // shipping bluesmirf with default of 9600
+        Serial.print("$$$");  // enter command mode
+        delay(100);
+        Serial.println(baudCmd);
+      }
+      Serial.begin(baud);
+    }
+
+    if (_mode == SERIAL_MODE_SOFTWARE || _mode == SERIAL_MODE_HARDWARE_AND_SOFTWARE) {
+      if (baud != 9600) {
+        _soft_serial->end();
+        _soft_serial->begin(9600);   // shipping bluesmirf with default of 9600
+        _soft_serial->print("$$$");  // enter command mode
+        delay(100);
+        _soft_serial->println(baudCmd);
+      }
+      _soft_serial->end();
+      _soft_serial->begin(baud);
+    }
+
+    if (_mode == SERIAL_MODE_HARDWARE_AND_SOFTWARE) {
+      Serial.begin(baud);
+    }
+  } else {
+    Serial.begin(9600);
+    Serial.print(baud);
+    Serial.println(" is not a supported bluetooth baud rate.");
+    Serial.println("Supported baud rates are:");
+    Serial.println("1200 2400 4800 9600 19200 38400 57600");
```

Comment:
Is there a preference between using the `F()` macro and verbosely creating a `PROGMEM` variable? Because I can't `Serial.print` a `PROGMEM` string without first creating a ~100 character buffer and `strcpy_P` to it.